src/hooks/gql.ts

```diff
@@ -45,8 +46,47 @@ export function useCurrentOrg() {
   }).data?.organizations_by_pk;
 }
 
-export function useCircleIdForEpoch(epochId: number) {
+export function useEpochIdForCircle(epochId: number) {
   return useTypedQuery(`circle-for-epoch-${epochId}`, {
-    epochs_by_pk: [{ id: epochId }, { circle_id: true }],
-  }).data?.epochs_by_pk?.circle_id;
+    epochs_by_pk: [
+      { id: epochId },
+      {
+        id: true,
+        ended: true,
+        circle_id: true,
+        number: true,
+        circle: {
+          id: true,
+          name: true,
+          users: [
+            { where: { received_gifts: { epoch_id: { _eq: epochId } } } },
```

Comment:
I think we shouldn't submit users who don't have gifts but, as for presentation, I'd suggest an order by Give so that you can use who's getting the most and then it tapers off naturally. I'm thinking about this in the context of a team member that has an undervalued contribution, just good for them to know when they are submitting so there are no surprises.